### PR TITLE
merge container and reference into one

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Removed:
- `balsamic init reference` and `balsamic init container`
- `--singularity` and `--reference-config` in `balsamic config case`
- 

Changed:
- `balsamic init` now does both `reference` and `container`. please see `balsamic init --help`

Added:
- `--balsamic-cache` and `--genome-version` to `balsamic config case` to handle input from `balsamic init`

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by hassanfa** [comment]
Tests are failing but that's story for another day! :-)

**Comment by ashwini06** [comment]
@hassanfa : Can you please expand a bit more on the description or attach screenshots of expected outcome directory looks like when you run `balsamic init`

**Comment by hassanfa** [comment]
Of course. I thought it is a draft, I'll fix those when draft is done.

**Comment by hassanfa** [comment]
balsamic init:
image

balsamic config removed and new options:
image

**Comment by hassanfa** [comment]
@ashwini06 ready to review.

File: BALSAMIC/commands/init/base.py [hunk 1]
```diff
@@ -1,21 +1,202 @@
...
+        LOG.info("Singularity image source: {}".format(container_stub_url))
+
+        # Set container name according to above docker image name
+        Path(container_outdir).mkdir(parents=True, exist_ok=True)
+        image_name = Path(container_outdir,
+                          "{}.sif".format(image_suffix)).as_posix()
+        LOG.info("Image will be downloaded to {}".format(image_name))
+        LOG.info("Starting download. This process can take some time...")
+
+        cmd = ["singularity", "pull", "--name", f"{image_name}"]
```
Comment (ashwini06) [inline BALSAMIC/commands/init/base.py:120]: 👍

File: BALSAMIC/commands/init/base.py [hunk 2]
```diff
@@ -1,21 +1,202 @@
...
+    dagfile_path = Path(reference_outdir, dagfile).as_posix()
+
+    config_dict["output"] = reference_outdir.as_posix()
+    if cosmic_key:
+        config_dict["cosmic_key"] = cosmic_key
+
+    config_dict["genome_version"] = genome_version
+
+    print(config_dict)
+    write_json(config_dict, config_json)
```
Comment (ashwini06) [inline BALSAMIC/commands/init/base.py:152]: 👍

**Review by ashwini06 — COMMENTED** [review]
Good to go :sparkles: 🚀   tested downloading of containers and reference, both parts work fine 👍

**Comment by hassanfa** [comment]
Example command to download reference and containers: `balsamic init -o balsamic_cache --cosmic-key "${COSMIC_KEY}" --genome-version hg19 -v develop ` for future reference.

**Pull request merged**